src/App.tsx

```diff
@@ -16,35 +17,103 @@ export const goodsFromServer = [
 ];
 
 export const App: React.FC = () => {
+  enum SortType {
+    ALPHABETICALLY,
+    BY_LENGTH,
+    RESET,
+  }
+
+  const [visibleList, setVisibleList] = useState<string[]>(goodsFromServer);
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+  const [actualSortType, setActualSortType] = useState<SortType>(
+    SortType.RESET,
+  );
+
+  const sortBy = (sortingType: SortType) => {
+    let sortedList: string[] = [];
+
+    switch (sortingType) {
+      case SortType.ALPHABETICALLY:
+        sortedList = [...goodsFromServer].sort((item1, item2) =>
+          item1.localeCompare(item2),
+        );
+        break;
+      case SortType.BY_LENGTH:
+        sortedList = [...goodsFromServer].sort(
+          (item1, item2) => item1.length - item2.length,
+        );
+        break;
+      case SortType.RESET:
+        sortedList = [...goodsFromServer];
+        setIsReversed(false);
+        break;
+    }
+
+    if (isReversed && sortingType !== SortType.RESET) {
+      sortedList.reverse();
+    }
+
+    setVisibleList(sortedList);
+    setActualSortType(sortingType);
+  };
+
+  const reverseList = () => {
```

Comment:
this part repeated, you can change your function for only one check isReversed

Reply:
I'm not sure I fully understand this one. Could you please explain a little bit more, would you kindly?  I think you are seeing something important at this point. And I'm not there yet))

Reply:
BTW - it is ok to have separate functions just to set the state, but performing actual manipulations on data is another thing